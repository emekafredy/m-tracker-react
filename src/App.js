import React, { Component } from 'react';
import { BrowserRouter, Route } from 'react-router-dom';
import { Provider } from "react-redux";
import jwt_decode from 'jwt-decode';

import Navbar from './components/Layout/Navbar';
import Footer from './components/Layout/Footer';
import WelcomeBody from './components/Layout/WelcomeBody';
import SignUp from './components/Auth/SignUp';
import Login from './components/Auth/Login';
import Requests from './components/User/Requests';
import CreateRequest from './components/User/CreateRequest';

// redux store configuration
import configureStore from './store/configureStore';

// utility function to set token
import setAuthToken from './utils/setAuthToken';
import { setCurrentUser, logUserOut } from './actions/auth';

// styling
import '../css/styles.css';

const store = configureStore();

// Check for authentication
if (localStorage.jwtToken) {
  setAuthToken(localStorage.jwtToken);
  const decoded = jwt_decode(localStorage.jwtToken);
  store.dispatch(setCurrentUser(decoded));

  // Check for expired token
  const currentTime = Date.now() / 1000;
  if (decoded.exp < currentTime) {
    store.dispatch(logUserOut());
    window.location.href = '/';
  }
}

class App extends Component {
  render() {
    return (
      <Provider store={ store }>
        <BrowserRouter>
          <div>
            <Navbar />
            <Route exact path='/' component={ WelcomeBody }/>
            <Route path='/register' component={ SignUp }/>
            <Route path='/login' component={ Login }/>
            <Route path="/requests" component={ Requests } />
            <Route path="/request/new" component={ CreateRequest } />
            <Footer />
          </div>
        </BrowserRouter>
      </Provider>
    )
  }
}

export default App;
