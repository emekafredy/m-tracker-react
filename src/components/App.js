import React, { Component } from 'react';

import Navbar from './Layout/Navbar';
import Footer from './Layout/Footer';
import AppRoutes from './AppRoutes';

class App extends Component {
  render() {
    return (
      <div>
        <Navbar />
        <AppRoutes />
        <Footer />
      </div>
    )
  }
}

export default App;
