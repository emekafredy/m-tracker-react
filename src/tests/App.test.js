import React from 'react';
import { shallow } from 'enzyme';
import { Provider } from 'react-redux';
import App from '../App';

describe('Signup component', () => {
  test("renders the signup component", () => {
    const wrapper = shallow(
    <Provider>
      <App />
    </Provider>
    );

    expect(wrapper.exists()).toBe(true);
  });
})
