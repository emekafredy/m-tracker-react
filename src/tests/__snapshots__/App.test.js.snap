// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders the routes in the App component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <BrowserRouter>
        <div>
          <Connect(withRouter(Navbar)) />
          <Switch>
            <Route
              component={[Function]}
              exact={true}
              path="/"
            />
            <Route
              component={[Function]}
              exact={true}
              path="/register"
            />
            <Route
              component={[Function]}
              exact={true}
              path="/login"
            />
            <Connect(PrivateRoute)
              component={[Function]}
              exact={true}
              path="/requests"
            />
            <Connect(PrivateRoute)
              component={[Function]}
              exact={true}
              path="/new"
            />
            <Connect(PrivateRoute)
              component={[Function]}
              exact={true}
              path="/request/:requestId"
            />
            <Connect(PrivateRoute)
              component={[Function]}
              exact={true}
              path="/request/:requestId/edit"
            />
            <Route
              component={[Function]}
            />
          </Switch>
          <Footer />
        </div>
      </BrowserRouter>,
      "store": Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <div>
          <Connect(withRouter(Navbar)) />
          <Switch>
            <Route
              component={[Function]}
              exact={true}
              path="/"
            />
            <Route
              component={[Function]}
              exact={true}
              path="/register"
            />
            <Route
              component={[Function]}
              exact={true}
              path="/login"
            />
            <Connect(PrivateRoute)
              component={[Function]}
              exact={true}
              path="/requests"
            />
            <Connect(PrivateRoute)
              component={[Function]}
              exact={true}
              path="/new"
            />
            <Connect(PrivateRoute)
              component={[Function]}
              exact={true}
              path="/request/:requestId"
            />
            <Connect(PrivateRoute)
              component={[Function]}
              exact={true}
              path="/request/:requestId/edit"
            />
            <Route
              component={[Function]}
            />
          </Switch>
          <Footer />
        </div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Connect(withRouter(Navbar)) />,
            <Switch>
              <Route
                component={[Function]}
                exact={true}
                path="/"
              />
              <Route
                component={[Function]}
                exact={true}
                path="/register"
              />
              <Route
                component={[Function]}
                exact={true}
                path="/login"
              />
              <Connect(PrivateRoute)
                component={[Function]}
                exact={true}
                path="/requests"
              />
              <Connect(PrivateRoute)
                component={[Function]}
                exact={true}
                path="/new"
              />
              <Connect(PrivateRoute)
                component={[Function]}
                exact={true}
                path="/request/:requestId"
              />
              <Connect(PrivateRoute)
                component={[Function]}
                exact={true}
                path="/request/:requestId/edit"
              />
              <Route
                component={[Function]}
              />
            </Switch>,
            <Footer />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <Route
                  component={[Function]}
                  exact={true}
                  path="/"
                />,
                <Route
                  component={[Function]}
                  exact={true}
                  path="/register"
                />,
                <Route
                  component={[Function]}
                  exact={true}
                  path="/login"
                />,
                <Connect(PrivateRoute)
                  component={[Function]}
                  exact={true}
                  path="/requests"
                />,
                <Connect(PrivateRoute)
                  component={[Function]}
                  exact={true}
                  path="/new"
                />,
                <Connect(PrivateRoute)
                  component={[Function]}
                  exact={true}
                  path="/request/:requestId"
                />,
                <Connect(PrivateRoute)
                  component={[Function]}
                  exact={true}
                  path="/request/:requestId/edit"
                />,
                <Route
                  component={[Function]}
                />,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "component": [Function],
                  "exact": true,
                  "path": "/",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "component": [Function],
                  "exact": true,
                  "path": "/register",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "component": [Function],
                  "exact": true,
                  "path": "/login",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "component": [Function],
                  "exact": true,
                  "path": "/requests",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "component": [Function],
                  "exact": true,
                  "path": "/new",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "component": [Function],
                  "exact": true,
                  "path": "/request/:requestId",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "component": [Function],
                  "exact": true,
                  "path": "/request/:requestId/edit",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "component": [Function],
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <BrowserRouter>
          <div>
            <Connect(withRouter(Navbar)) />
            <Switch>
              <Route
                component={[Function]}
                exact={true}
                path="/"
              />
              <Route
                component={[Function]}
                exact={true}
                path="/register"
              />
              <Route
                component={[Function]}
                exact={true}
                path="/login"
              />
              <Connect(PrivateRoute)
                component={[Function]}
                exact={true}
                path="/requests"
              />
              <Connect(PrivateRoute)
                component={[Function]}
                exact={true}
                path="/new"
              />
              <Connect(PrivateRoute)
                component={[Function]}
                exact={true}
                path="/request/:requestId"
              />
              <Connect(PrivateRoute)
                component={[Function]}
                exact={true}
                path="/request/:requestId/edit"
              />
              <Route
                component={[Function]}
              />
            </Switch>
            <Footer />
          </div>
        </BrowserRouter>,
        "store": Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <div>
            <Connect(withRouter(Navbar)) />
            <Switch>
              <Route
                component={[Function]}
                exact={true}
                path="/"
              />
              <Route
                component={[Function]}
                exact={true}
                path="/register"
              />
              <Route
                component={[Function]}
                exact={true}
                path="/login"
              />
              <Connect(PrivateRoute)
                component={[Function]}
                exact={true}
                path="/requests"
              />
              <Connect(PrivateRoute)
                component={[Function]}
                exact={true}
                path="/new"
              />
              <Connect(PrivateRoute)
                component={[Function]}
                exact={true}
                path="/request/:requestId"
              />
              <Connect(PrivateRoute)
                component={[Function]}
                exact={true}
                path="/request/:requestId/edit"
              />
              <Route
                component={[Function]}
              />
            </Switch>
            <Footer />
          </div>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Connect(withRouter(Navbar)) />,
              <Switch>
                <Route
                  component={[Function]}
                  exact={true}
                  path="/"
                />
                <Route
                  component={[Function]}
                  exact={true}
                  path="/register"
                />
                <Route
                  component={[Function]}
                  exact={true}
                  path="/login"
                />
                <Connect(PrivateRoute)
                  component={[Function]}
                  exact={true}
                  path="/requests"
                />
                <Connect(PrivateRoute)
                  component={[Function]}
                  exact={true}
                  path="/new"
                />
                <Connect(PrivateRoute)
                  component={[Function]}
                  exact={true}
                  path="/request/:requestId"
                />
                <Connect(PrivateRoute)
                  component={[Function]}
                  exact={true}
                  path="/request/:requestId/edit"
                />
                <Route
                  component={[Function]}
                />
              </Switch>,
              <Footer />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  <Route
                    component={[Function]}
                    exact={true}
                    path="/"
                  />,
                  <Route
                    component={[Function]}
                    exact={true}
                    path="/register"
                  />,
                  <Route
                    component={[Function]}
                    exact={true}
                    path="/login"
                  />,
                  <Connect(PrivateRoute)
                    component={[Function]}
                    exact={true}
                    path="/requests"
                  />,
                  <Connect(PrivateRoute)
                    component={[Function]}
                    exact={true}
                    path="/new"
                  />,
                  <Connect(PrivateRoute)
                    component={[Function]}
                    exact={true}
                    path="/request/:requestId"
                  />,
                  <Connect(PrivateRoute)
                    component={[Function]}
                    exact={true}
                    path="/request/:requestId/edit"
                  />,
                  <Route
                    component={[Function]}
                  />,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "component": [Function],
                    "exact": true,
                    "path": "/",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "component": [Function],
                    "exact": true,
                    "path": "/register",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "component": [Function],
                    "exact": true,
                    "path": "/login",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "component": [Function],
                    "exact": true,
                    "path": "/requests",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "component": [Function],
                    "exact": true,
                    "path": "/new",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "component": [Function],
                    "exact": true,
                    "path": "/request/:requestId",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "component": [Function],
                    "exact": true,
                    "path": "/request/:requestId/edit",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "component": [Function],
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
